backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/UserAuthorizationVerifier.java

```diff
@@ -21,21 +27,69 @@
+    private OktaRepository _oktaRepo;
 
-    public UserAuthorizationVerifier(SiteAdminEmailList admins, IdentitySupplier supplier, OrganizationService orgService) {
+    public UserAuthorizationVerifier(SiteAdminEmailList admins, 
+                                     IdentitySupplier supplier, 
+                                     OrganizationService orgService,
+                                     ApiUserRepository userRepo,
+                                     OktaRepository oktaRepo) {
         super();
         this._admins = admins;
         this._supplier = supplier;
         this._orgService = orgService;
+        this._userRepo = userRepo;
+        this._oktaRepo = oktaRepo;
     }
 
     public boolean userHasSiteAdminRole() {
         IdentityAttributes id = _supplier.get();
-        return id != null && _admins.contains(id.getUsername());
+        if (id == null) {
+            return false;
+        }
+        getUser(id.getUsername());
+        return _admins.contains(id.getUsername());
     }
 
     public boolean userHasPermission(UserPermission permission) {
+        IdentityAttributes id = _supplier.get();
+        if (id == null) {
+            return false;
+        }
+        getUser(id.getUsername());
         Optional<OrganizationRoles> orgRoles = _orgService.getCurrentOrganizationRoles();
         return orgRoles.isPresent() && orgRoles.get().getGrantedPermissions().contains(permission);
     }
+
+    public boolean userIsInSameOrg(UUID userId) {
+        Optional<OrganizationRoles> currentOrgRoles = _orgService.getCurrentOrganizationRoles();
+        String otherUserEmail = getUser(userId).getLoginEmail();
+        Optional<Organization> otherOrg = _oktaRepo.getOrganizationRoleClaimsForUser(otherUserEmail)
+                .map(r -> _orgService.getOrganization(r.getOrganizationExternalId()));
+        return currentOrgRoles.isPresent() && otherOrg.isPresent() 
+                ? currentOrgRoles.get().getOrganization().getExternalId().equals(otherOrg.get().getExternalId())
+                : false;
+    }
+
+    // This replicates getUser() in ApiUserService.java, but we cannot call that logic directly or else that method
+    // would have to either a) become public with no method-level security, which is bad; or b) become public with
+    // method-level security that invokes an above method which creates a circular loop with this method.
+    private ApiUser getUser(UUID id) {
+        Optional<ApiUser> found = _userRepo.findById(id);
+        if (!found.isPresent()) {
+            throw new NonexistentUserException();
+        }
+        ApiUser user = found.get();
+        return user;
+    }
+
+    private ApiUser getUser(String loginEmail) {
+        Optional<ApiUser> found = _userRepo.findByLoginEmail(loginEmail);
+        if (!found.isPresent()) {
+            throw new NonexistentUserException();
+        }
+        ApiUser user = found.get();
+        return user;
+    }
```

Comment:
To reduce duplication and since the returns user is not used I would rewrite this and then update the calls in `userHasSiteAdminRole` and `userHasPermission`

```suggestion
    private void hasValidIdentity() {
        IdentityAttributes id = _supplier.get();
        if (id == null) {
            throw new NonexistentUserException();
        }
        Optional<ApiUser> found = _userRepo.findByLoginEmail(id.getUsername());
        if (!found.isPresent()) {
            throw new NonexistentUserException();
        }
    }
```